currency_monthly_rate/models/res_currency.py

```diff
@@ -0,0 +1,95 @@
+# Copyright 2018 Camptocamp SA
+# License AGPL-3.0 or later (https://www.gnu.org/licenses/agpl).
+
+from odoo import models, fields, api, _
+
+
+class ResCurrency(models.Model):
+    _inherit = "res.currency"
+
+    monthly_rate = fields.Float(compute='_compute_current_monthly_rate',
+                                string='Current Monthly Rate', digits=(12, 6),
+                                help='The monthly rate of the currency to '
+                                     'the currency of rate 1.')
+    monthly_rate_ids = fields.One2many('res.currency.rate.monthly',
+                                       'currency_id', string='Monthly rates')
+
+    @api.multi
+    def _compute_current_monthly_rate(self):
+        date = self.env.context.get('date') or fields.Date.today()
+        company_id = self.env.context.get('company_id') or self.env[
+            'res.users']._get_company().id
+        # the subquery selects the first rate before 'date' for the given
+        # currency/company
+        query = """SELECT c.id, (SELECT r.rate
+                                 FROM res_currency_rate_monthly r
+                                 WHERE r.currency_id = c.id
+                                 AND r.name <= %s
+                                 AND (r.company_id IS NULL
+                                      OR r.company_id = %s)
+                                 ORDER BY r.company_id, r.name DESC
+                                 LIMIT 1) AS rate
+                   FROM res_currency c
+                   WHERE c.id IN %s"""
+        self._cr.execute(query, (date, company_id, tuple(self.ids)))
+        currency_rates = dict(self._cr.fetchall())
+        for currency in self:
+            currency.monthly_rate = currency_rates.get(currency.id) or 1.0
+
+    @api.model
+    def _get_conversion_rate(self, from_currency, to_currency):
+        monthly = self.env.context.get('monthly_rate')
+        if not monthly:
+            return super()._get_conversion_rate(from_currency, to_currency)
```

Comment:
Today I learned that is not required class and object for py3.5 in super method `super(ResCurrency, self)` 🙇